### PR TITLE
🌱 Added basic integration test

### PR DESCRIPTION
**Pull request by huutomerkki**
This test will only create a cluster and collect logs from it. The resource flavor will be fixed here https://github.com/metal3-io/project-infra/pull/554

Fixes: #897

### CODE REVIEW

File: test/e2e/basic_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		validateGlobals(specName)
+
+		// We need to override clusterctl apply log folder to avoid getting our credentials exposed.
+		clusterctlLogFolder = filepath.Join(os.TempDir(), "clusters", bootstrapClusterProxy.GetName())
+	})
+
+	It("Should create a workload cluster", func() {
+		By("Fetching cluster configuration")
+		numberOfWorkers = int(*e2eConfig.GetInt32PtrVariable("WORKER_MACHINE_COUNT"))
+		numberOfControlplane = int(*e2eConfig.GetInt32PtrVariable("CONTROL_PLANE_MACHINE_COUNT"))
```
Comment (furkatgofurov7) [inline test/e2e/basic_integration_test.go:25]: where are these vars being used?

Comment (huutomerkki) [reply test/e2e/basic_integration_test.go:25]: I copied them over from the integration test, they are not used there either. Only place where they're read seems to be in pivoting_based_feature_test.go.

Comment (huutomerkki) [reply test/e2e/basic_integration_test.go]: Removed them.

Comment (mboukhalfa) [reply test/e2e/basic_integration_test.go]: Are not they used in `createTargetCluster` ?

Comment (mboukhalfa) [reply test/e2e/basic_integration_test.go]: seems that will get from e2e_suite https://github.com/metal3-io/cluster-api-provider-metal3/blob/660431812eea1a522e37940eddd1ef600095bd5f/test/e2e/e2e_suite_test.go#L117

**Review by furkatgofurov7 — COMMENTED** [review]
Thanks @huutomerkki, looks good to me!

Small nit on PR description: 

> Related issue: #897

I think this has to be changed to: `Fixed: #` in order to close the issue automatically once this is merged. 

/lgtm

**Comment by furkatgofurov7** [comment]
> /hold
> 
> to fix the description (feel free to unhold when it is ready :))

Should be fine now.

/hold cancel

**Pull request merged**